socorro/scripts/fetch_crashids.py

```diff
@@ -29,17 +31,56 @@
 
 HOST = 'https://crash-stats.mozilla.com'
 
+MAX_PAGE = 1000
 
-def fetch_crashids(params):
+
+def fetch_crashids(params, num_results):
+    """Generator that returns crash ids
+
+    :arg dict params: dict of super search parameters to base the query on
+    :arg varies num: "all" or a number of results to get
+
+    :returns: generator of crash ids
+
+    """
     url = HOST + '/api/SuperSearch/'
 
-    resp = requests.get(url, params)
-    if resp.status_code == 200:
+    # Set up retrying on 429s
+    retries = Retry(total=32, backoff_factor=1, status_forcelist=[429])
+    session = requests.Session()
+    session.mount('http://', HTTPAdapter(max_retries=retries))
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+
+    # Set up first page
+    params['_results_offset'] = 0
+    params['_results_number'] = MAX_PAGE if num_results == 'all' else min(MAX_PAGE, num_results)
+
+    # Fetch pages of crash ids until we've gotten as many as we want or there aren't any more to get
+    crashids_count = 0
+    while True:
+        resp = session.get(url, params=params)
+        if resp.status_code != 200:
+            raise Exception('Bad response: %s %s' % (resp.status_code, resp.content))
+
         hits = resp.json()['hits']
-        crashids = [hit['uuid'] for hit in hits]
-        return crashids
 
-    raise Exception('Bad response: %s %s' % (resp.status_code, resp.content))
+        for hit in hits:
+            crashids_count += 1
+            yield hit['uuid']
+
+            # If we've gotten as many crashids as we need, we return
+            if num_results != 'all' and crashids_count >= num_results:
+                return
+
+        # If there are no more crash ids to get, we return
+        total = resp.json()['total']
+        if not hits or crashids_count >= total:
+            return
+
+        # Get the next page
+        params['_results_offset'] += MAX_PAGE
+        if num_results != 'all':
+            params['_results_number'] = min(MAX_PAGE, total - crashids_count)
```

Comment:
It's possible there's some off-by-one shenanigans in here. Maybe one of the `>=` should be a `>`. It's hard to tell. I thought about adding tests for this, but they'd have to be heavily mocked and at that point, it didn't seem worth it.

Someone else can write those tests if we change our minds.